docs/architecture/adr-036-arbitrary-signature.md

```diff
@@ -0,0 +1,112 @@
+
+
+## Decision
+
+The aim is being able to sign arbitrary messages, even using Ledger or similar HSM devices.
+
+As a result signed messages should look roughly like Cosmos SDK messages but **must not** be a valid on-chain transaction. `chain-id`, `account_number` and `sequence` can all be assigned invalid values.
+
+Cosmos SDK 0.40 also introduces a concept of “auth_info” this can specify SIGN_MODES.
+
+A spec should include an `auth_info` that supports SIGN_MODE_DIRECT and SIGN_MODE_LEGACY_AMINO.
+
+An offchain transaction follows these rules:
+
+- the memo must be empty
+- nonce, sequence number must be equal to 0
+- chain-id must be equal to “”
+- fee gas must be equal to 0
+- fee amount must be an empty array
+- inside the message with type `MsgSignData`, we put *bytes* data and the `address` of the signer.
+- it's applications developers decision how `Data` should be treated, by treated we mean the serialization and deserialization process and the Object `Data` should represent. 
+
+Verification of an offchain transaction follows the same rules as an onchain one, except for the spec differences highlighted above.
+
+Proto definition:
+```proto
+// MsgSignData defines an arbitrary, general-purpose, off-chain message
+message MsgSignData {
+    // Signer is the sdk.AccAddress of the message signer
+    bytes Signer = 1 [(gogoproto.jsontag) = "signer", (gogoproto.casttype) = "github.com/cosmos/cosmos-sdk/types.AccAddress"];
+    // Data represents the raw bytes of the content that is signed (text, json, etc)
+    bytes Data = 2 [(gogoproto.jsontag) = "data"];
+}
+```
+Signed MsgSignData json example:
+```json
+{
+  "type": "cosmos-sdk/StdTx",
+  "value": {
+    "msg": [
+      {
+        "type": "sign/MsgSignData",
+        "value": {
+          "signer": "cosmos1hftz5ugqmpg9243xeegsqqav62f8hnywsjr4xr",
+          "data": "cmFuZG9t"
+        }
+      }
+    ],
+    "fee": {
+      "amount": [],
+      "gas": "0"
+    },
+    "signatures": [
+      {
+        "pub_key": {
+          "type": "tendermint/PubKeySecp256k1",
+          "value": "AqnDSiRoFmTPfq97xxEb2VkQ/Hm28cPsqsZm9jEVsYK9"
+        },
+        "signature": "8y8i34qJakkjse9pOD2De+dnlc4KvFgh0wQpes4eydN66D9kv7cmCEouRrkka9tlW9cAkIL52ErB+6ye7X5aEg=="
+      }
+    ],
+    "memo": ""
+  }
+}
+```
+
+## Consequences
+
+There is a specification on how messages, that are not meant to be broadcast to a live chain, should be formed. 
+
+### Backwards Compatibility
+
+Backwards compatibility is maintained as this is a new message spec definition.
+
+### Positive
+
+- A common format that can be used by multiple applications to sign and verify off-chain messages.
+- The specification is primitive which means it can cover every use case without limiting what is possible to fit inside it.
+- It gives room for other off-chain messages specifications that aim to target more specific and common use cases such as off-chain-based authN/authZ layers [2].
+
+### Negative
+
```

Comment:
```suggestion

+ Current proposal requires a fixed relation between account address and a public key.
+ Doesn't work with multi sig accounts.
```

Reply:
Please feel free to add more negative consequences - we were discussing it in the thread. Leaving it empty is not good.